crates/swc_plugin_runner/src/cache.rs

```diff
@@ -0,0 +1,153 @@
+use std::{
+    env::current_dir,
+    path::{Path, PathBuf},
+};
+
+use anyhow::{Context, Error};
+use parking_lot::Mutex;
+use swc_common::{
+    collections::AHashMap,
+    sync::{Lazy, OnceCell},
+};
+use wasmer::{Module, Store};
+use wasmer_cache::{Cache as WasmerCache, FileSystemCache, Hash};
+
+/// Version for bytecode cache stored in local filesystem.
+///
+/// This MUST be updated when bump up wasmer.
+///
+/// Bytecode cache generated via wasmer is generally portable,
+/// however it is not gauranteed to be compatible across wasmer's
+/// internal changes.
+/// https://github.com/wasmerio/wasmer/issues/2781
+const MODULE_SERIALIZATION_VERSION: &str = "v1";
```

Comment:
So, it is reasonably safe to believe bytecode caches are portable (i.e share between CI machines). What we need to ensure is as wrote should invalidate cache when we bump up wasmer. 

The tricky part is when we `expire` those caches. We do not delete caches ever at the moment, so cache will indefinitely grow and users have to manage it on their own. Maybe we may not need to worry that much as other tools basically do the same.